lib/io/spi.hpp

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include <cstdint>
+#include <cstring>
+
+#include <core/logger.hpp>
+#include <core/types.hpp>
+
+#if __linux__
```

Comment:
is this a system macro?

Reply:
this is from the old code except the old one would not actually detect linux. since I'm already defining the relevant stuff from spidev, would this even be necessary?